src/parser/deserializer.ts

```diff
@@ -45,6 +45,19 @@ export interface DeserializeOptions {
   index?: number;
 
   raw?: boolean;
+  /** allows for opt-out utf-8 validation for all keys or
+   * specified keys. Must be all true or all false.
```

Comment:
Let's also add type tests for this property (we probably want to add a new file, since the existing one just tests the actual BSON types; this one would test input to deserialize)